### PR TITLE
Added support for localized day-of-week (weekdays and first day of we…

### PR DESCRIPTION
**Pull request by ankraft**

### CODE REVIEW

File: TEAChart/TEAContributionGraph.m [hunk 1]
```diff
@@ -99,15 +99,17 @@ - (void)drawRect:(CGRect)rect
     CGContextRef context = UIGraphicsGetCurrentContext();
     
     NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:TEACalendarIdentifierGregorian];
+    [calendar setLocale:[NSLocale currentLocale]];
```
Comment (xhacker) [inline TEAChart/TEAContributionGraph.m]: I’d suggest using dot notation here (`calendar.locale = ...`).

File: TEAChart/TEAContributionGraph.m [hunk 2]
```diff
@@ -99,15 +99,17 @@ - (void)drawRect:(CGRect)rect
...
     NSDate *firstDay = [calendar dateFromComponents:comp];
-    
+    NSUInteger firstDayOfWeek = [calendar firstWeekday];
+	
     comp.month = comp.month + 1;
     NSDate *nextMonth = [calendar dateFromComponents:comp];
-    
-    NSArray *weekdayNames = @[@"S", @"M", @"T", @"W", @"T", @"F", @"S"];
-    
+
```
Comment (xhacker) [inline TEAChart/TEAContributionGraph.m]: Can you avoid making white space changes?

File: TEAChart/TEAContributionGraph.m [hunk 3]
```diff
@@ -99,15 +99,17 @@ - (void)drawRect:(CGRect)rect
     CGContextRef context = UIGraphicsGetCurrentContext();
     
     NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:TEACalendarIdentifierGregorian];
+    [calendar setLocale:[NSLocale currentLocale]];
     NSDateComponents *comp = [calendar components:(NSCalendarUnitYear | NSCalendarUnitMonth | NSCalendarUnitDay) fromDate:_graphMonth];
     comp.day = 1;
     NSDate *firstDay = [calendar dateFromComponents:comp];
-    
+    NSUInteger firstDayOfWeek = [calendar firstWeekday];
```
Comment (xhacker) [inline TEAChart/TEAContributionGraph.m]: Also dot notation.

File: TEAChart/TEAContributionGraph.m [hunk 4]
```diff
@@ -119,7 +121,7 @@ - (void)drawRect:(CGRect)rect
             NSFontAttributeName: [UIFont fontWithName:@"HelveticaNeue-Light" size:self.cellSize * 0.65],
             NSParagraphStyleAttributeName: paragraphStyle,
         };
-        [weekdayNames[i] drawInRect:rect withAttributes:attributes];
+        [weekdayNames[(i+firstDayOfWeek-1)%7] drawInRect:rect withAttributes:attributes];
```
Comment (xhacker) [inline TEAChart/TEAContributionGraph.m]: Can you add spaces beside operators?

**Comment by xhacker** [comment]
Thanks for contribution! Just a few comments.

Also, can you tell me a locale that starts from Monday so that I can test?

**Comment by ankraft** [comment]
Thanks for the review. I made the changes as  you requested. I hope this commit covers all your comments.

Regarding the locale: try Germany.

**Comment by xhacker** [comment]
Thanks! Tried Germany and it worked perfectly.

Just two more places that could use dot notation: `veryShortWeekdaySymbols` and `[calendar setFirstWeekday:firstWeekday];`. Also please don’t change spaces to tabs in line 110 and 112. Then I’ll merge.

**Comment by xhacker** [comment]
Perfect, thank you!

**Comment by ankraft** [comment]
Done. Xcode automatically converted the spaces to tabs when I saved the file. Should be okay now, I hope. Thank you for your support.

**Pull request merged**